### ISSUE TITLE
It's unclear how to implement authentication

### ISSUE THREAD

**Issue by mac9416** [argoproj-labs/argo-client-java#4]
The bottom of the README mentions which authentication methods are available, but it provides neither 1) an explanation of configuring authentication in argo-server nor 2) code examples for how to use authentication once it's set up.

Since the README is generated, we may need to use some OpenAPI-Generator config to augment the README. Or alternatively, we may need to create an additional file documenting authentication usage.

**Comment by mac9416** [comment]
Finally found an example [here](https://github.com/argoproj-labs/argo-client-gen/blob/f2203ea835ce5c60e4ea373b54d19466ce3d23b1/java-test/src/test/java/io/argoproj/workflow/ArgoServerAPI.java). Would it make sense to move that into this repo?

**Comment by unikzforce** [comment]
Yeah, It would help a lot of people. at least It could be mentioned in the README.md .
